Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.35552&attempt_id=90dff4f2-8252-440e-a748-7d074510cda1)

<details><summary>Partial log (click to expand)</summary><p>

```
make[1]: *** [Makefile:28: bspatch.o] Error 127
make[1]: Leaving directory '/private/tmp/nix-build-imgpatchtools-0.3.drv-0/source/applypatch'
/nix/store/pkjmwq7sqrvjg7cjiph6hq0khsmfl6p8-bash-4.4-p12/bin/bash: line 1: cd: android-base: No such file or directory
/nix/store/pkjmwq7sqrvjg7cjiph6hq0khsmfl6p8-bash-4.4-p12/bin/bash: line 1: cd: edify: No such file or directory
/nix/store/pkjmwq7sqrvjg7cjiph6hq0khsmfl6p8-bash-4.4-p12/bin/bash: line 1: cd: minzip: No such file or directory
/nix/store/pkjmwq7sqrvjg7cjiph6hq0khsmfl6p8-bash-4.4-p12/bin/bash: line 1: cd: otafault: No such file or directory
/nix/store/pkjmwq7sqrvjg7cjiph6hq0khsmfl6p8-bash-4.4-p12/bin/bash: line 1: cd: blockimg: No such file or directory
make: *** [Makefile:31: sub] Error 1
builder for '/nix/store/bayjryya5rjzhrvb4ikdd1pp0hh9zvj2-imgpatchtools-0.3.drv' failed with exit code 2
error: build of '/nix/store/bayjryya5rjzhrvb4ikdd1pp0hh9zvj2-imgpatchtools-0.3.drv' failed
```
</p></details>

--- a/pkgs/development/mobile/imgpatchtools/default.nix
+++ b/pkgs/development/mobile/imgpatchtools/default.nix
@@ -1,0 +1,30 @@
+{ stdenv, fetchzip, bzip2, openssl, zlib }:
+
+stdenv.mkDerivation rec {
+  name = "imgpatchtools-${version}";
+  version = "0.3";
+
+  src = fetchzip {
+    url = "https://github.com/erfanoabdi/imgpatchtools/archive/${version}.tar.gz";
+    sha256 = "1cwp1hfhip252dz0mbkhrsrkws6m15k359n4amw2vfnglnls8czd";
+  };
+
+  buildInputs = [ bzip2 openssl zlib ];
+
+  installPhase = "install -Dt $out/bin bin/*";
+
+  meta = with stdenv.lib; {
+    description = "Tools to manipulate Android OTA archives";
+    longDescription = ''
+      This package is useful for Android development. In particular, it can be
+      used to extract ext4 /system image from Android distribution ZIP archives
+      such as those distributed by LineageOS and Replicant, via BlockImageUpdate
+      utility. It also includes other, related, but arguably more advanced tools
+      for OTA manipulation.
+    '';
+    homepage = https://github.com/erfanoabdi/imgpatchtools;
+    license = licenses.gpl3;
+    maintainers = with maintainers; [ yegortimoshenko ];
+    platforms = platforms.linux;
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -644,6 +644,8 @@ with pkgs;
 
   glyr = callPackage ../tools/audio/glyr { };
 
+  imgpatchtools = callPackage ../development/mobile/imgpatchtools { };
+
   lastpass-cli = callPackage ../tools/security/lastpass-cli { };
 
   pass = callPackage ../tools/security/pass { };

